test/sys/test_event.rs

```diff
@@ -0,0 +1,41 @@
+use libc::intptr_t;
+use nix::sys::event::{EventFilter, EventFlag, FilterFlag, KEvent};
+
+#[test]
+fn test_struct_kevent() {
+    use std::mem;
+
+    let udata: intptr_t = 12345;
+    let data: intptr_t = 0x1337;
+
+    let actual = KEvent::new(
+        0xdead_beef,
+        EventFilter::EVFILT_READ,
+        EventFlag::EV_ONESHOT | EventFlag::EV_ADD,
+        FilterFlag::NOTE_CHILD | FilterFlag::NOTE_EXIT,
+        data,
+        udata,
+    );
+    assert_eq!(0xdead_beef, actual.ident());
+    assert_eq!(EventFilter::EVFILT_READ, actual.filter().unwrap());
```

Comment:
Originally, this line of code was:

```rs
let filter = actual.kevent.filter;
assert_eq!(libc::EVFILT_READ, filter);
```

But the `kevent` field (libc::kevent) of  `KEvent` is private, so I changed it to the corresponding public method `.filter()`